pyjobshop/ProblemData.py

```diff
@@ -2,7 +2,7 @@
 from copy import deepcopy
 from dataclasses import dataclass
 from enum import Enum
-from typing import Optional, TypeVar
+from typing import Optional, Sequence, TypeVar
```

Comment:
List is not covariant, so we have to use Sequence. This is getting fairly complex in typing so I'm in doubt whether this is the right direction to go into.